### PR TITLE
Add provider json configuration, adding a dummy Coverr provider

### LINKED ISSUE #2804: Add provider configuration json fragment
## Task Description

Add a json fragment to describe each media provider type:

```
[{
  type, // ‘unsplash’, ‘coverr’, ‘tenor’ - capitalize to convert to 
        // the api ProviderType enum.
  displayName,
  supportedContentTypes, // array of ‘image’ | ’video’
  requiresAuthorAttribution,
  supportsOrderByLastUpdateTime,
}]
```

Where practical, provider specific details should be configured here, to avoid difficult to manage if/else statements scattered throughout the code.

Add constants to a sensible place, and replace 'local' in media/local/reducers|actions with a constant.

As per state design doc:
https://docs.google.com/document/d/1S9jTh6Lz3UEhj9Wkb7S7x_kJGvRVpEaNOqk2aCyb4y8#heading=h.mywdzjii4j8

Also:

Add a new ProviderTab for the Coverr provider.
In onProviderTabClick, call setSelectedProvider to update the selected provider.
Verify keyboard navigation.
Similar to this, though it needs cleaning up:
https://github.com/google/web-stories-wp/pull/3364/files#diff-dd6e2e686434fe0300a8809ad6ad354aR213

Also...

For now, fetching of categories is only supported by Unsplash:
https://github.com/google/web-stories-wp/pull/3359/files#diff-e43e03f8e5a52dac0cd4e124a50b8ec0R113

Update useFetchCategoriesEffect to call listCategories() only for Unsplash to avoid throwing an error and showing a snackbar.

### LINKED ISSUE #3349: Add a feature flag for Coverr
## Task Description

Add 'coverrTab' feature flag to:
https://github.com/google/web-stories-wp/blob/main/includes/Story_Post_Type.php

### LINKED ISSUE #3355: Add Coverr attribution to the bottom of the Coverr provider tab
## Task Description

This will work in the same way as Unsplash attribution, and can be found in the figma design.

The new Coverr logo can be found here:
White: https://drive.google.com/file/d/1o883-bRgXb_BnMgsb_rKgg6U4cYa275y
Dark: https://drive.google.com/file/d/1z0KoEvnWcJnqpIj0y_Kwqqgxo2P5JI_l

eg.
https://share.getcloudapp.com/7KumKkbZ

### PR DESCRIPTION
**Pull request by diegovar**
## Summary

Refactors most of the media pane logic to depend on a set of provided json configs. Adds a dummy coverr provider that currently just delegates to Unsplash with a modified query.

Screen Shot 2020-08-05 at 1 24 02 pm

## Relevant Technical Choices

The Coverr "Flag" is in providerConfiguration.js because the providers list is referenced inside hooks, so I can't inject useFeatures to access our normal flag map there (as it would mean using a hook inside another hook).

## To-do

Could extract the per-provider error messages into the configuration as well.

## User-facing changes

Adds a new provider, makes them configurable.

## Testing Instructions

Enable media3pTab` flag and set `showCoverrTab` in `providerConfiguration.js` to true.

---

Fixes #2804
Fixes #3349
Fixes #3355

### CODE REVIEW

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  new Map(
+    [
+      [
+        ProviderType.UNSPLASH,
+        {
+          displayName: 'Unsplash',
+          supportedContentTypes: [ContentType.IMAGE],
+          supportsCategories: true,
+          requiresAuthorAttribution: true,
+          attributionBuilder: UnsplashAttribution,
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/providerConfiguration.js]: [nit]: Since this is a component, should this be named attributionComponent or attributionPill?

File: assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js [hunk 2]
```diff
@@ -67,9 +59,20 @@ function Media3pApiProvider({ children }) {
     selectedCategoryId,
     mediaType,
   }) {
-    if (provider.toLowerCase() !== Providers.UNSPLASH) {
+    if (
+      provider !== ProviderType.UNSPLASH &&
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js]: Perhaps this could be:
```
if (!Object.keys(Providers).contains(provider)) {
  throw ...;
}
```

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    <a href={'https://coverr.co'} target={'_blank'} rel={'noreferrer'}>
+      <AttributionPill>
+        {__('Powered by', 'web-stories')}
+        <span style={LOGO_STYLE}>{'COVERR'}</span>
+      </AttributionPill>
+    </a>
+  );
+}
+
+const showCoverrTab = false;
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/providerConfiguration.js]: To keep the features in useFeatures(), perhaps the name of the flag can be added as an attribute?
```
featureName: 'showCoverrTab'
```

File: assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js [hunk 4]
```diff
@@ -60,7 +60,7 @@ describe('useFetchMediaEffect', () => {
         useFetchMediaEffect({
           provider: 'unsplash',
           selectedProvider: 'unsplash',
-          pageToken: 'pageToken',
+          pageToken: undefined,
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js:63]: Why was this part changed? This was testing that the current page's token is propagated through fetchMediaStart, fetchMediaSuccess etc.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+/**
+ *
+ * @type {Object.<string, ProviderConfiguration>}
+ */
+export const Providers = Object.fromEntries(
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/providerConfiguration.js]: [nit]: 'Providers' returns a variable not a class, so it should have a lowercase 'p'. Or perhaps it makes sense to be the default export?

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+ * @property {string} displayName The display name of the provider.
+ * @property {ContentType[]} supportedContentTypes The supported content types by the provider.
+ * @property {boolean} supportsCategories Whether this provider supports filtering media by category.
+ * @property {boolean} requiresAuthorAttribution Whether this provider requires showing author
+ * attribution on each media element.
+ * @property {function(): React.Component} attributionBuilder A function that constructs an attribution
+ * React Component for this provider.
+ */
+
+function UnsplashAttribution() {
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/providerConfiguration.js]: This file is rendering AttributionPill and providing providerConfiguration, perhaps the AttributionPill parts belong in a separate file?

File: assets/src/edit-story/app/media/media3p/providerReducer.js [hunk 7]
```diff
@@ -17,7 +17,7 @@
 /**
  * Internal dependencies
  */
-import commonReducer, {
+import paginationReducer, {
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/providerReducer.js:20]: Thanks :)

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js [hunk 8]
```diff
@@ -27,12 +27,15 @@ import { useEffect } from 'react';
...
-import { ProviderType } from '../../../components/library/panes/media/common/providerType';
+import { ProviderType } from '../providerType';
+import usePrevious from '../usePrevious';
 import { useMedia3pApi } from './api';
 
 function getFetchMediaErrorMessage(provider) {
   if (provider === ProviderType.UNSPLASH) {
     return __('Error loading media from Unsplash', 'web-stories');
+  } else if (provider === ProviderType.COVERR) {
+    return __('Error loading media from Coverr', 'web-stories');
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js]: The display names ('Unsplash' or 'Coverr') should be read from the provider configuration.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+      ],
+      showCoverrTab
+        ? [
+            ProviderType.COVERR,
+            {
+              displayName: 'Coverr',
+              supportedContentTypes: [ContentType.VIDEO],
+              supportsCategories: false,
+              requiresAuthorAttribution: false,
+              attributionBuilder: CoverrAttribution,
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Why did you not just put the properties for the attribution here instead of the whole component?

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+
+/**
+ *
+ * @type {Object.<string, ProviderConfiguration>}
+ */
+export const Providers = Object.fromEntries(
+  new Map(
+    [
+      [
+        ProviderType.UNSPLASH,
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/media3p/providerConfiguration.js]: I think we should remove the whole `ProviderType` construct and instead just define a unique provider key in here. 

Everywhere where we currently use a check whether the provider matches a certain `ProviderType` should be avoided and be part of the configuration.

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js [hunk 11]
```diff
@@ -77,49 +97,47 @@ function Media3pPane(props) {
...
-  const {
-    state: { categories },
-    actions: { selectCategory, deselectCategory },
-  } = selectedProviderState;
-
   useEffect(() => {
-    if (isActive) {
-      setSelectedProvider({ provider: 'unsplash' });
+    if (isActive && !selectedProvider) {
+      setSelectedProvider({ provider: ProviderType.UNSPLASH });
```
Comment (winterDroid) [inline assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: Should it be whatever is first in the providers config?

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js [hunk 12]
```diff
@@ -77,49 +97,47 @@ function Media3pPane(props) {
...
 
-  const {
-    state: { categories },
-    actions: { selectCategory, deselectCategory },
-  } = selectedProviderState;
-
   useEffect(() => {
-    if (isActive) {
-      setSelectedProvider({ provider: 'unsplash' });
+    if (isActive && !selectedProvider) {
```
Comment (cvolzke4) [inline assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js:121]: What's the reason for the !selectedProvider check?

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+        {
+          displayName: 'Unsplash',
+          supportedContentTypes: [ContentType.IMAGE],
+          supportsCategories: true,
+          requiresAuthorAttribution: true,
+          attributionBuilder: UnsplashAttribution,
+        },
+      ],
+      showCoverrTab
+        ? [
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/media3p/providerConfiguration.js]: nit: can't this be `showCoverrTab && [...]`

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js (hunk 12 above)
Comment (cvolzke4) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js:121]: Nevermind, I understand it now :)

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js [hunk 14]
```diff
@@ -130,32 +148,65 @@ function Media3pPane(props) {
...
+            return (
+              <ProviderMediaCategoriesWrapper
+                dataProvider={providerType}
+                {...wrapperProps}
+                key={`provider-bottom-wrapper-${providerType}`}
+              >
+                {Providers[providerType].supportsCategories && (
+                  <Media3pCategories
+                    categories={
+                      media3p[providerType].state.categories.categories
```
Comment (cvolzke4) [inline assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: Could media3p[providerType].state / edia3p[providerType].actions be extracted to local variables?
```
<Media3pCategories
    categories={providerState.categories.categories}
    selectedCategoryId={providerState.categories.selectedCategoryId}
    selectCategory={providerActions.selectCategory}
    deselectCategory={providerActions.deselectCategory}
```

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js [hunk 15]
```diff
@@ -197,20 +172,7 @@ function PaginatedMediaGallery({
...
-          target={'_blank'}
-          rel={'noreferrer'}
-        >
-          <AttributionPill>
-            {__('Powered by', 'web-stories')}
-            <UnsplashLogoFull style={LOGO_PROPS} />
-          </AttributionPill>
-        </a>
-      )}
+      {Providers[providerType].attributionBuilder()}
```
Comment (cvolzke4) [inline assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js]: To make it clear what 'attributionBuilder' is, I would suggest:
```
  const AttributionComponent = Providers[providerType].attributionBuilder;
  return (
    <>
      <MediaGalleryContainer
        data-testid="media-gallery-container"
        ref={refCallbackContainer}
      >
        <MediaGalleryInnerContainer>{mediaGallery}</MediaGalleryInnerContainer>
      </MediaGalleryContainer>
      <AttributionComponent />
    </>
  );
```

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 5 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js:105]: The config can then be specified as a simple object:
```
export const providers = {
  [ProviderType.UNSPLASH]: {
    displayName: 'Unsplash',
    supportedContentTypes: [ContentType.IMAGE],
    supportsCategories: true,
    requiresAuthorAttribution: true,
    attributionBuilder: UnsplashAttribution,
  },
  [ProviderType.COVERR]: {
    displayName: 'Coverr',
    supportedContentTypes: [ContentType.VIDEO],
    supportsCategories: false,
    requiresAuthorAttribution: false,
    attributionBuilder: CoverrAttribution,
  },
}
```

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 3 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: The config can then be specified as a simple object:
```
export const providers = {
  [ProviderType.UNSPLASH]: {
    displayName: 'Unsplash',
    supportedContentTypes: [ContentType.IMAGE],
    supportsCategories: true,
    requiresAuthorAttribution: true,
    attributionBuilder: UnsplashAttribution,
  },
  [ProviderType.COVERR]: {
    displayName: 'Coverr',
    supportedContentTypes: [ContentType.VIDEO],
    supportsCategories: false,
    requiresAuthorAttribution: false,
    attributionBuilder: CoverrAttribution,
    featureName: 'showCoverrTab',
  },
}
```

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js [hunk 16]
```diff
@@ -53,41 +57,66 @@ export default function useFetchMediaEffect({
...
-    if (provider === selectedProvider) {
+    // If we switched provider tab, and we already had loaded a page there, we
+    // don't load media again.
+    const somethingChanged =
+      previousProps &&
+      (pageToken != previousProps.pageToken ||
+        searchTerm != previousProps.searchTerm ||
+        selectedCategoryId != previousProps.selectedCategoryId);
+    const firstFetchOrSomethingChanged =
+      (!pageToken && !nextPageToken) || somethingChanged;
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js]: I don't think you need 'nextPageToken' to know if it's the first fetch. pageToken will be undefined if the first page is being fetched.

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js [hunk 17]
```diff
@@ -53,41 +57,66 @@ export default function useFetchMediaEffect({
...
+
+    if (provider === selectedProvider && firstFetchOrSomethingChanged) {
       fetch();
     }
+    // We don't want to depend on previousProps, see https://blog.logrocket.com/how-to-get-previous-props-state-with-react-hooks/
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js:109]: Another way to avoid depending on previousProps, you could use the previousProps reference inside of the useEffect block:

```
  const previousPropsRef = useRef();
  useEffect(() => {
    const previousProps = previousPropsRef.current;
    previousPropsRef.current = { pageToken, searchTerm, selectedCategoryId };
    // ...
  }, [
    // The ref never changes
    previousPropsRef
  ]);
```

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 1 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Done

File: assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js (hunk 2 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js]: Done.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 3 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Good idea!

File: assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js (hunk 4 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js:63]: So with this current implementation, the media is only fetched if no media has been fetched (pageToken and nextPageToken are both falsey) or sth has changed from the previous render. I modified the test to ensure that the fetch would happen, and I wasn't able to figure out how to trigger the case where dependencies of a useEffect hook change. Do you know how to do that?

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 5 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: I think it's the same as providerType.js right? Not sure why it would be lowercase p?

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 6 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Done.

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js (hunk 8 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js]: I couldn't find a way of doing i18n message interpolation using __ (seems very basic), so I've moved the messages to the config.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 9 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Because we might need different styles of attribution per provider. Maybe Unsplash is ok with a "Powered by <SVG>" but Tenor wants something different.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 10 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: I'd rather do that in a separate PR, that might be quite a bit of work and this PR is big enough.

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js (hunk 11 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: Good catch.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 13 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: I've moved to use featureName so we can have it as a feature flag.

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js (hunk 14 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: Done.

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js (hunk 15 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js]: I renamed the field to attributionComponent so it should be clear now.

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js (hunk 16 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js]: I think if you've already fetched data, nextPageToken is not null but pageToken is. We don't want to fetch if we've already fetched in that case, so I need to check both?

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js (hunk 17 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js:109]: I'll try this out and report back.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 10 above)
Comment (winterDroid) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Yeah that's fair 👍

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 9 above)
Comment (winterDroid) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: That makes sense. But we can define what needs to be customized and provided. It does not just need to be one image or so.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 10 above)
Comment (winterDroid) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Do you want to create a follow-up ticket for it or do it in the same? If latter one, this probably should just address and not fix it according to the summary.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 5 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: ProviderType is an enum, whereas this is now a field:
go/js-style#naming-enum-names
go/js-style#naming-non-constant-field-names

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js (hunk 16 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js]: ahh, you are right.
I wonder if that's a problem, ie. if there is <= 1 page of images, then pageToken and nextPageToken will both be undefined, even if the first fetch has already completed.

File: assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js (hunk 4 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js:63]: Did you try calling renderUseFetchMediaEffect() again with a different property value for one of the dependencies? I think that will re-trigger the useEffect.

Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js:63]: I did, it re-triggers the useEffect but the previous props are not kept at all. I suspect because it's a completely different instance.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 5 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: Given this is a constant name, I've renamed it to PROVIDERS.

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 10 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: I've created #3707

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js (hunk 16 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js]: Good point, I've modified the logic to depend on isMediaLoading and isMediaLoaded instead. If both are false, that means that no media has been fetched yet.

File: assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js (hunk 4 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js:63]: Interesting, perhaps useRef is only maintained within the same act() call?

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js (hunk 17 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js:109]: I don't think this would do what we want. Accessing previousPropsRef.current inside the effect would not give you the previous value of the props as it's updated inside an effect in usePrevious.

File: assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js (hunk 4 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js:63]: yeah ,tried running the useFetchMediaEffect in the same act, and even inside the same renderHook, but the useRef is not maintained :(

File: assets/src/edit-story/app/media/media3p/providerConfiguration.js (hunk 9 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/providerConfiguration.js]: I think I'll leave this as is, and in the meantime will confirm with Tenor and Unsplash if this is the attribution format they both expect.

File: assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js [hunk 18]
```diff
@@ -67,9 +60,17 @@ function Media3pApiProvider({ children }) {
...
     mediaType,
   }) {
-    if (provider.toLowerCase() !== Providers.UNSPLASH) {
+    if (!Object.keys(PROVIDERS).includes(provider)) {
       throw new Error(`Unsupported provider: ${provider}`);
     }
+
+    // Temporary hack alert!: Convert coverr to unsplash for testing until
+    // Coverr backend is implemented.
+    if (provider === ProviderType.COVERR) {
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js:69]: Refer to a ticket.

File: assets/src/edit-story/app/media/media3p/attribution.js [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+  left: 24px;
+  bottom: 10px;
+  border-radius: 100px;
+  padding: 5px 8px;
+  line-height: 16px;
+  display: flex;
+  flex-wrap: nowrap;
+  font-size: 12px;
+  color: ${theme.colors.fg.white};
+  background-color: ${rgba(theme.colors.bg.black, 0.7)};
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/media3p/attribution.js]: In most places we use `${({ theme }) => theme.fonts.duration.family}` and then don't import the `theme` directly

File: assets/src/edit-story/app/media/media3p/useFetchCategoriesEffect.js [hunk 20]
```diff
@@ -54,21 +47,30 @@ export default function useFetchCategoriesEffect({
...
       fetchCategoriesStart({ provider });
       try {
-        const { categories } = await listCategories({ provider });
-        fetchCategoriesSuccess({ provider, categories });
-      } catch {
+        const { categories: newCategories } = await listCategories({
+          provider,
+        });
+        fetchCategoriesSuccess({ provider, categories: newCategories });
+      } catch (e) {
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/media3p/useFetchCategoriesEffect.js:54]: The 'e' is unused here.

File: assets/src/edit-story/app/media/usePrevious.js [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+import { useEffect, useRef } from 'react';
+
+/**
+ * React hook to access the previous value of props or state in a hook.
+ * See https://blog.logrocket.com/how-to-get-previous-props-state-with-react-hooks/
+ *
+ * @param {Object} value Object to keep track of.
+ * @return {Object} The previous value of the object.
+ */
+export default function usePrevious(value) {
```
Comment (winterDroid) [inline assets/src/edit-story/app/media/usePrevious.js]: Shall we have a test for that?

File: assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js [hunk 22]
```diff
@@ -197,20 +172,7 @@ function PaginatedMediaGallery({
...
-          target={'_blank'}
-          rel={'noreferrer'}
-        >
-          <AttributionPill>
-            {__('Powered by', 'web-stories')}
-            <UnsplashLogoFull style={LOGO_PROPS} />
-          </AttributionPill>
-        </a>
-      )}
+      {PROVIDERS[providerType].attributionComponent()}
```
Comment (winterDroid) [inline assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js]: I think calling it a builder did make more sense as it build the component.

File: assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js (hunk 17 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js:109]: The usePrevious hook would not be used in the example given above.
previousPropsRef.current is updated in the useFetchMediaEffect's useEffect().

**Review by cvolzke4 — APPROVED** [review]
Looking good!

**Review by winterDroid — APPROVED** [review]

File: assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js (hunk 18 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js:69]: Done

File: assets/src/edit-story/app/media/media3p/attribution.js (hunk 19 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/media3p/attribution.js]: Done.

File: assets/src/edit-story/app/media/usePrevious.js (hunk 21 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/usePrevious.js]: Removes this file.
